src/main/java/seedu/address/logic/commands/util/EditElderlyDescriptor.java

```diff
@@ -0,0 +1,100 @@
+     */
+    public EditElderlyDescriptor(EditElderlyDescriptor toCopy) {
+        super(toCopy);
+        setRiskLevel(toCopy.riskLevel);
+    }
+
+    /**
+     * Creates and returns a {@code Elderly} with the details of {@code elderlyToEdit}
+     * edited with {@code editElderlyDescriptor}.
+     */
+    public static Elderly createEditedElderly(Elderly elderlyToEdit, EditPersonDescriptor editPersonDescriptor) {
+        assert elderlyToEdit != null;
+
+        Name updatedName = editPersonDescriptor.getName().orElse(elderlyToEdit.getName());
+        Phone updatedPhone = editPersonDescriptor.getPhone().orElse(elderlyToEdit.getPhone());
+        Email updatedEmail = editPersonDescriptor.getEmail().orElse(elderlyToEdit.getEmail());
+        Address updatedAddress = editPersonDescriptor.getAddress().orElse(elderlyToEdit.getAddress());
+        Nric updatedNric = editPersonDescriptor.getNric().orElse(elderlyToEdit.getNric());
+        Age updatedAge = editPersonDescriptor.getAge().orElse(elderlyToEdit.getAge());
+        Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(elderlyToEdit.getTags());
+        // Setting elderly-specific attributes
+        RiskLevel updatedRiskLevel = elderlyToEdit.getRiskLevel();
+        if (editPersonDescriptor instanceof EditElderlyDescriptor) {
+            EditElderlyDescriptor editElderlyDescriptor = (EditElderlyDescriptor) editPersonDescriptor;
+            updatedRiskLevel = editElderlyDescriptor.getRiskLevel()
+                    .orElse(elderlyToEdit.getRiskLevel());
+        }
+        return new Elderly(updatedName, updatedPhone, updatedEmail, updatedAddress,
+                updatedNric, updatedAge, updatedRiskLevel, updatedTags);
+    }
+
+
+    /**
+     * Returns true if at least one field is edited.
+     */
+    public boolean isAnyFieldEdited() {
+        return super.isAnyFieldEdited() || CollectionUtil.isAnyNonNull(riskLevel);
+    }
+
+    public void setRiskLevel(RiskLevel riskLevel) {
+        this.riskLevel = riskLevel;
+    }
+
+    public Optional<RiskLevel> getRiskLevel() {
+        return Optional.ofNullable(riskLevel);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof EditElderlyDescriptor)) {
+            return false;
+        }
+
+        // state check
+        EditElderlyDescriptor e = (EditElderlyDescriptor) other;
+
+        return getName().equals(e.getName())
+                && getPhone().equals(e.getPhone())
+                && getEmail().equals(e.getEmail())
+                && getAddress().equals(e.getAddress())
+                && getNric().equals(e.getNric())
+                && getAge().equals(e.getAge())
+                && getRiskLevel().equals(e.getRiskLevel())
+                && getTags().equals(e.getTags());
+    }
+}
```

Comment:
Later after this round of merging was done, I may need to do further adjustments to the EditDescriptor classes